### ISSUE TITLE
Add Chairman's Worksheet Notes to Customize Lists

### ISSUE THREAD

**Issue by stecchio66** [ajtruckle/meeting-schedule-assistant-support#7]
I need to add notes for the chairman for all midweek meetings. So, I add this text for each week:

`Announcements:<ul><li>Announcement 1</li><li>Announcement 2</li></ul>`

The result is this:

Announcements:

- Announcement 1
- Announcement 2

Is it possible to add custom text for chairman notes to the Customize List window?

**Comment by ajtruckle** [comment]
Hi @stecchio66 

I am sure we can do something for you. Question: Do you need the ability to add announcements to **both** Opening / Closing Comments?

This would make it as flexible as possible.

**Comment by stecchio66** [comment]
If you want you can add this for both comments.
As you say it would be more flexible.

**Comment by ajtruckle** [comment]
@stecchio66 
I have been thinking about implementation and this is my proposal:

1. We add **Announcement** as a category to the [Customize Lists](https://help-msa.publictalksoftware.co.uk/mwb-editor-customize-lists.html) window as you suggested.
2. We add two new menu buttons to the [Chairman's Worksheet](https://help-msa.publictalksoftware.co.uk/mwb-editor-chairman-notes.html) window:
   - The _button_ text will be **Announcements**.
   - The _menu item_ text will be **Manage**.
3. Clicking the **Announcements** button will display a new popup window. This window will have a checklist of all available announcements. You can choose the ones you want to include on the worksheet. These announcements will be saved in the MWB data files.
4. Clicking the **Manage** menu item will display the **Customize Lists** window for you to add / edit / delete announcements.
5. We could possibly (not sure 🤔 ) have a **multi-line tooltip** that shows what the selected announcements are. The tooltip would display when you hover the mouse over the Announcements menu button.
6. We could also possibly have (not sure 🤔) two icons:
   - An icon to indicate that _one or more announcements_ have been selected.
   - Another icon to indicate that _no announcements_ have been selected.
7. The editor will add the announcements into the XML data file. For example:
```lang xml
<OpeningComments>
    <Theme>Opening Comments</Theme>
    <Time StartTime12="7:20" StartTime24="19:20" EndTime12="7:21" EndTime24="19:21" DurationShort="1 min.">1 min.</Time>
	<Announcements>
		<Announcement>Announcement 1.</Announcement>
		<Announcement>Announcement 2.</Announcement>
		<Announcement>Announcement 3.</Announcement>
	</Announcements>	
</OpeningComments>
```
8. Finally, the default **Worksheet** templates can be updated to to display these announcements in a bulleted list.

Thoughts? 🤔

**Comment by stecchio66** [comment]
You always think bigger. You're too good.
I would have content with much less but what your thought is exceptional.

**Comment by ajtruckle** [comment]
@stecchio66 
Thanks for the kind words.😀 I have decided to simplify the popup window design:

- There will be no controls to change the order of the announcements.
- All announcements will be sorted in A - Z instead. 

I came to this decision because it would not be possible to main a different sort order week by week. So keeping the announcements sorted in A to Z seems the simplest to implement.

**Comment by ajtruckle** [comment]
@stecchio66 
I have made another change. When you edit the announcement list for a given week it now add any announcements that are not in the list back into the list and sets them checked. Does that make sense? 😖 Just in case you delete an announcement that you are still using for other weeks.

**Comment by ajtruckle** [comment]
@stecchio66
I have sent you a new beta (privately) to test. Two default templates have been updated:

- Worksheet
- Worksheet RTL

Here is a snapshot of the new features in action:

![Meeting Editor](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/3cbb7861-72e0-4ae3-b7c8-8cb0ecc380be)

The announcements are displayed on their own row on the schedule.

**Comment by fstouma** [comment]
Announcements option is working fine for Arabic

**Comment by stecchio66** [comment]
@ajtruckle 
the last beta work very well.
This is some example of announcements:

- Announcements and Reminders (October 2023)
- Monthly Congregation Accounts Report
- Assembly Preview
- Convention Preview
- Changes to the next midweek meeting
- Auxiliary Pioneers of the month

**Comment by ajtruckle** [comment]
@stecchio66 @fstouma 
Thanks for doing the beta testing brothers! 🙏 I'll use those sample announcements in the help documentation.

I have decided to re-work the **Chairman's Notes** window:

![Chairman's Notes window](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/450234f7-4841-4fa6-80d7-83de048e92d8)

Benefits:

- Changed the name of the first button to **Reminders**.  This is because _announcements_ are usually given during the _concluding comments_ of the meeting.
- Moved the **Concluding Comments** to the bottom right. This is the logical place. Thanks for the private feedback @cengizu.

**Comment by ajtruckle** [comment]
This feature is now implemented and is available in version 24.1.2.

> [!NOTE]  
>  The [online newsletter](https://www.publictalksoftware.co.uk/2023/10/msa-version-24-1-2-news/) for version **24.1.2** is now available.
